instrumentation/src/javatests/org/oppia/android/instrumentation/topic/TopicActivityTest.kt

```diff
@@ -0,0 +1,136 @@
+package org.oppia.android.instrumentation.topic
+
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import com.google.common.truth.Truth.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.findObjectByRes
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.findObjectByText
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.scrollRecyclerViewTextIntoView
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.startOppiaFromScratch
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.waitForRes
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.waitForText
+
+/** Tests for TopicActivity. */
+class TopicActivityTest {
+  private lateinit var device: UiDevice
+
+  @Before
+  fun setUp() {
+    // Initialize UiDevice instance
+    device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+    device.startOppiaFromScratch()
+  }
+
+  @Test
+  fun testTopicActivity_infoFragment_clickSeeMore_showSeeLess() {
+    navigateToTopicActivity()
+    device.findObjectByText("See More").click()
+    // Assert See Less is visible
+    val seeLessText = device.findObjectByText("See Less")
+    assertThat(seeLessText).isNotNull()
+  }
+
+  @Test
+  fun testTopicActivity_infoFragment_clickSeeLess_showSeeMore() {
+    navigateToTopicActivity()
+    device.findObjectByText("See More").click()
+    device.waitForText("See Less")
+    device.findObjectByText("See Less").click()
+    // Assert See Less is visible
+    val seeMoreText = device.findObjectByText("See More")
+    assertThat(seeMoreText).isNotNull()
+  }
```

Comment:
@ayush0402 since we only need to check whether all the components of the TopicActivity are working we don't need multiple test cases for this.

The flow is exactly same as the video you sent in the chat. make sure to attach a video of the running tests.

Reply:
include these back, but in the single testcase.

Reply:
PTAL if this is the correct implementation for the test. Also, updated the video.

Reply:
Done.